2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/ScalaCollectorApp.scala

```diff
@@ -200,11 +204,42 @@ class CollectorConfig(config: Config) {
   val minBackoff = backoffPolicy.getLong("minBackoff")
   val maxBackoff = backoffPolicy.getLong("maxBackoff")
 
-  private val kafka = sink.getConfig("kafka")
-  val kafkaBrokers = kafka.getString("brokers")
-  private val kafkaTopic = kafka.getConfig("topic")
-  val kafkaTopicGoodName = kafkaTopic.getString("good")
-  val kafkaTopicBadName = kafkaTopic.getString("bad")
+  object Kafka {
+    private val kafkaConfig = sink.getConfig("kafka")
+
+    object Topic {
+      private val topicConfig = kafkaConfig.getConfig("topic")
+      val topicGoodName: String = topicConfig.getString("good")
+      val topicBadName: String = topicConfig.getString("bad")
+    }
+
+    // get producer related configurations
+    object Producer {
+      private val producerConfig = kafkaConfig.getConfig("producer")
+      assert(producerConfig.hasPath("bootstrap.servers"), "bootstrap.servers is required")
+
+      private val defaultProperties: Properties = {
+        val props = new Properties()
+        props.put("acks", "all")
+        props.put("buffer.memory", byteLimit.toString)
+        props.put("batch.size", recordLimit.toString)
```

Comment:
Ah true, that's bytes. byteLimit.toString then, right? And buffer.memory removed?